config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -158,75 +158,100 @@ private void removeEntitiesForDeleteBundle(final List<Entity> entities) {
-            annotatedPolicyMap.put(policyNameWithPath, policyEntity);
-            Set<Dependency> dependencies = policyEntity.getUsedEntities();
-            List<GatewayEntity> excludedReusableEntities = new ArrayList<>();
-            if (dependencies != null) {
-                for (Dependency dependency : dependencies) {
-                    Class<? extends GatewayEntity> entityClass = entityTypeRegistry.getEntityClass(dependency.getType());
-                    Map<String, ? extends GatewayEntity> allEntitiesOfType = bundle.getEntities(entityClass);
-                    Optional<? extends Map.Entry<String, ? extends GatewayEntity>> optionalGatewayEntity =
-                            allEntitiesOfType.entrySet().stream()
-                            .filter(e -> {
-                                GatewayEntity gatewayEntity = e.getValue();
-                                if (gatewayEntity.getName() != null) {
-                                    return dependency.getName().equals(gatewayEntity.getName());
-                                } else {
-                                    return dependency.getName().equals(PathUtils.extractName(e.getKey()));
-                                }
-                            }).findFirst();
-                    Map<String, ? extends GatewayEntity> entityMap = annotatedBundle.getEntities(entityClass);
-                    optionalGatewayEntity.ifPresent(entry -> insertGatewayEntity(entityMap, entry.getKey(),
-                            entry.getValue(), excludeReusable, excludedReusableEntities));
+            loadPolicyDependencies(policyEntity, annotatedBundle, rawBundle, excludeReusable);
+        }
+    }
+
+    /**
+     * Loads the Policy and its dependencies
+     *
+     * @param policy Policy for which gateway dependencies needs to be loaded.
+     * @param annotatedBundle Annotated Bundle for which bundle is being created.
+     * @param rawBundle Bundle containing all the entities of the gateway.
+     * @param excludeReusable Exclude loading Reusable entities as the dependencies of the policy
+     */
+    private void loadPolicyDependencies(Policy policy, AnnotatedBundle annotatedBundle, Bundle rawBundle,
+                                                    boolean excludeReusable) {
+        if (excludeGatewayEntity(Policy.class, policy, annotatedBundle, excludeReusable)) {
+            return;
+        }
+
+        loadFolderDependencies(annotatedBundle, policy);
+
+        Map<String, Policy> annotatedPolicyMap = annotatedBundle.getEntities(Policy.class);
+        annotatedPolicyMap.put(policy.getPath(), policy);
+
+        Set<Dependency> dependencies = policy.getUsedEntities();
+        if (dependencies != null) {
+            for (Dependency dependency : dependencies) {
+                switch (dependency.getType()) {
+                    case EntityTypes.POLICY_TYPE:
+                        Policy dependentPolicy = rawBundle.getPolicies().get(dependency.getName());
```

Comment:
rawBundle Policy map will have key as path, but dependency has name

Reply:
Fixed it!